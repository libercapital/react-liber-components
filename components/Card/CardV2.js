"use strict";

require("core-js/modules/es6.object.define-property");

require("core-js/modules/web.dom.iterable");

require("core-js/modules/es6.array.iterator");

require("core-js/modules/es6.object.to-string");

require("core-js/modules/es6.string.iterator");

require("core-js/modules/es6.weak-map");

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports.InnerCardHeader = exports.InnerCard = exports.CardFooter = exports.CardContent = exports.CardHeader = exports.Card = void 0;

var _styledComponents = _interopRequireWildcard(require("styled-components"));

var _Themes = require("../Themes");

function _getRequireWildcardCache() { if (typeof WeakMap !== "function") return null; var cache = new WeakMap(); _getRequireWildcardCache = function _getRequireWildcardCache() { return cache; }; return cache; }

function _interopRequireWildcard(obj) { if (obj && obj.__esModule) { return obj; } var cache = _getRequireWildcardCache(); if (cache && cache.has(obj)) { return cache.get(obj); } var newObj = {}; if (obj != null) { var hasPropertyDescriptor = Object.defineProperty && Object.getOwnPropertyDescriptor; for (var key in obj) { if (Object.prototype.hasOwnProperty.call(obj, key)) { var desc = hasPropertyDescriptor ? Object.getOwnPropertyDescriptor(obj, key) : null; if (desc && (desc.get || desc.set)) { Object.defineProperty(newObj, key, desc); } else { newObj[key] = obj[key]; } } } } newObj["default"] = obj; if (cache) { cache.set(obj, newObj); } return newObj; }

var getTheme = function getTheme(key, def) {
  return (0, _styledComponents.css)(["", ""], function (props) {
    return (0, _Themes.handleThemeFromObject)(props.theme, key, def);
  });
};

var elevation = function elevation() {
  var level = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : '1';
  return {
    1: (0, _styledComponents.css)(["", ""], getTheme('box-shadows.1', (0, _styledComponents.css)(["box-shadow:0 1px 2px 0 rgba(64,95,113,0.12);"]))),
    2: (0, _styledComponents.css)(["", ""], getTheme('box-shadows.2', (0, _styledComponents.css)(["box-shadow:0 2px 4px 0 rgba(64,95,113,0.12);"]))),
    3: (0, _styledComponents.css)(["", ""], getTheme('box-shadows.3', (0, _styledComponents.css)(["box-shadow:0 4px 12px 0 rgba(64,95,113,0.12);"])))
  }[level];
};

var highlight = function highlight() {
  var color = arguments.length > 0 && arguments[0] !== undefined ? arguments[0] : 'default';
  return {
    "default": (0, _styledComponents.css)(["transparent"]),
    dark: (0, _styledComponents.css)(["", ""], getTheme('colors.dark-text', '#405f71')),
    primary: (0, _styledComponents.css)(["", ""], getTheme('colors.primary', '#009dff')),
    success: (0, _styledComponents.css)(["", ""], getTheme('colors.success', '#13ce66')),
    alert: (0, _styledComponents.css)(["", ""], getTheme('colors.alert', '#fabe00')),
    error: (0, _styledComponents.css)(["", ""], getTheme('colors.error', '#dc3545'))
  }[color];
};

var Card = _styledComponents["default"].div.withConfig({
  displayName: "CardV2__Card",
  componentId: "mom78k-0"
})(["", " border-radius:4px;background-color:", ";min-width:50px;min-height:50px;display:flex;flex-direction:column;justify-content:flex-start;align-items:center;"], function (props) {
  return elevation(props.elevation);
}, getTheme('colors.white', '#ffffff'));

exports.Card = Card;
var headerTg = (0, _styledComponents.css)(["font-size:16px;font-weight:500;line-height:22px;"]);

var CardHeader = _styledComponents["default"].div.withConfig({
  displayName: "CardV2__CardHeader",
  componentId: "mom78k-1"
})(["", " font-family:", ";color:", ";border-bottom:1px solid ", ";width:100%;height:48px;display:flex;align-items:center;justify-content:center;text-align:center;padding:0px 24px;box-sizing:border-box;position:relative;&:after{content:\"\";position:absolute;bottom:-1px;height:0px;left:24px;width:calc(100% - 48px);border-bottom:2px solid ", ";}"], getTheme('typography.subhead', headerTg), getTheme('font-family', 'Roboto'), getTheme('colors.dark-text', '#405f71'), getTheme('colors.border', '#e8ebee') || 'transparent', function (props) {
  return highlight(props.highlight);
});

exports.CardHeader = CardHeader;
var contentTg = (0, _styledComponents.css)(["font-size:16px;font-weight:400;line-height:22px;"]);

var CardContent = _styledComponents["default"].div.withConfig({
  displayName: "CardV2__CardContent",
  componentId: "mom78k-2"
})(["", " color:", ";flex-grow:1;padding:24px;"], getTheme('typography.body', contentTg), getTheme('colors.dar-text70', '#798e9b'));

exports.CardContent = CardContent;

var CardFooter = _styledComponents["default"].div.withConfig({
  displayName: "CardV2__CardFooter",
  componentId: "mom78k-3"
})(["", " font-family:", ";color:", ";border-top:1px solid ", ";width:100%;height:48px;display:flex;align-items:center;justify-content:center;text-align:center;"], getTheme('typography.subhead', headerTg), getTheme('font-family', 'Roboto'), getTheme('colors.dark-text38', '#b6c2c9'), getTheme('colors.border', '#e8ebee'));

exports.CardFooter = CardFooter;
var InnerCard = (0, _styledComponents["default"])(Card).withConfig({
  displayName: "CardV2__InnerCard",
  componentId: "mom78k-4"
})(["box-shadow:unset;border:1px solid ", ";background-color:", ";"], getTheme('colors.border', '#e8ebee'), getTheme('colors.header', '#f9fafa'));
exports.InnerCard = InnerCard;
var InnerCardHeader = (0, _styledComponents["default"])(CardHeader).withConfig({
  displayName: "CardV2__InnerCardHeader",
  componentId: "mom78k-5"
})(["justify-content:left;text-align:flex-start;font-weight:", ";"], getTheme('font-weights.regular', '400'));
exports.InnerCardHeader = InnerCardHeader;