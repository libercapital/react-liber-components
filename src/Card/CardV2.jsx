import styled, { css } from 'styled-components';
import { handleThemeFromObject } from '../Themes';

const getTheme = (key, def) => css`${props => handleThemeFromObject(props.theme, key, def)}`;


const elevation = (level = '1') => ({
  1: css`${getTheme('box-shadows.1', css`box-shadow: 0 1px 2px 0 rgba(64, 95, 113, 0.12);`)}`,
  2: css`${getTheme('box-shadows.2', css`box-shadow: 0 2px 4px 0 rgba(64, 95, 113, 0.12);`)}`,
  3: css`${getTheme('box-shadows.3', css`box-shadow: 0 4px 12px 0 rgba(64, 95, 113, 0.12);`)}`,
}[level]);

const highlight = (color = 'default') => ({
  default: css`transparent`,
  dark: css`${getTheme('colors.dark-text', '#405f71')}`,
  primary: css`${getTheme('colors.primary', '#009dff')}`,
  success: css`${getTheme('colors.success', '#13ce66')}`,
  alert: css`${getTheme('colors.alert', '#fabe00')}`,
  error: css`${getTheme('colors.error', '#dc3545')}`,
}[color]);

export const Card = styled.div`
  ${props => elevation(props.elevation)}
  border-radius: 4px;
  background-color: ${getTheme('colors.white', '#ffffff')};
  min-width: 50px;
  min-height: 50px;

  display: flex;
  flex-direction: column;
  justify-content: flex-start;
  align-items: center;
`;

const headerTg = css`
  font-size: 16px;
  font-weight: 500;
  line-height: 22px;
`;
export const CardHeader = styled.div`
  ${getTheme('typography.subhead', headerTg)}
  font-family: ${getTheme('font-family', 'Roboto')};
  color: ${getTheme('colors.dark-text', '#405f71')};

  border-bottom: 1px solid ${getTheme('colors.border', '#e8ebee') || 'transparent'};

  width: 100%;
  height: 48px;
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;
  padding: 0px 24px;
  box-sizing: border-box;

  position: relative;

  &:after{
    content: "";
    position: absolute;
    bottom: -1px;
    height: 0px;
    left: 24px;
    width: calc(100% - 48px);
    border-bottom: 2px solid ${props => highlight(props.highlight)};
  }
`;

const contentTg = css`
  font-size: 16px;
  font-weight: 400;
  line-height: 22px;
`;

export const CardContent = styled.div`
  ${getTheme('typography.body', contentTg)}
  color: ${getTheme('colors.dar-text70', '#798e9b')};
  flex-grow: 1;
  padding: 24px;
`;

export const CardFooter = styled.div`
  ${getTheme('typography.subhead', headerTg)}
  font-family: ${getTheme('font-family', 'Roboto')};
  color: ${getTheme('colors.dark-text38', '#b6c2c9')};

  border-top: 1px solid ${getTheme('colors.border', '#e8ebee')};

  width: 100%;
  height: 48px;
  display: flex;
  align-items: center;
  justify-content: center;
  text-align: center;

`;

export const InnerCard = styled(Card)`
  box-shadow: unset;
  border: 1px solid ${getTheme('colors.border', '#e8ebee')};
  background-color: ${getTheme('colors.header', '#f9fafa')};
`;

export const InnerCardHeader = styled(CardHeader)`
 justify-content: left;
 text-align: flex-start;
 font-weight: ${getTheme('font-weights.regular', '400')};
`;
