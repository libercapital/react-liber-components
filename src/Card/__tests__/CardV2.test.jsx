import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import 'jest-styled-components';
import * as styles from '../CardV2';

describe('Testing Card v2 Styles', () => {
  configure({ adapter: new Adapter() });
  const { Card, CardHeader, ...uncontrolled } = styles;
  describe('Uncontrolled', () => {
    Object.keys(uncontrolled).forEach((style) => {
      it(`renders ${style} correctly`, () => {
        const App = uncontrolled[style];
        const wrapper = shallow(<App />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });
  });
  describe('controlled', () => {
    describe('Card', () => {
      it('renders correctly without props', () => {
        const wrapper = shallow(<Card />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
      it('renders correctly without with elevation=1', () => {
        const wrapper = shallow(<Card elevation="1" />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
      it('renders correctly without with elevation=2', () => {
        const wrapper = shallow(<Card elevation="2" />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
      it('renders correctly without with elevation=3', () => {
        const wrapper = shallow(<Card elevation="3" />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
      it('renders correctly without with elevation=other', () => {
        const wrapper = shallow(<Card elevation="other" />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });
    describe('CardHeader', () => {
      it('renders correctly without props', () => {
        const wrapper = shallow(<CardHeader />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
      it('renders correctly without with highlight=default', () => {
        const wrapper = shallow(<Card highlight="default" />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
      it('renders correctly without with highlight=dark', () => {
        const wrapper = shallow(<Card highlight="dark" />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
      it('renders correctly without with highlight=primary', () => {
        const wrapper = shallow(<Card highlight="primary" />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
      it('renders correctly without with highlight=success', () => {
        const wrapper = shallow(<Card highlight="success" />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
      it('renders correctly without with highlight=alert', () => {
        const wrapper = shallow(<Card highlight="alert" />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
      it('renders correctly without with highlight=error', () => {
        const wrapper = shallow(<Card highlight="error" />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
      it('renders correctly without with highlight=other', () => {
        const wrapper = shallow(<Card highlight="other" />);
        expect(toJson(wrapper)).toMatchSnapshot();
      });
    });
  });
});
