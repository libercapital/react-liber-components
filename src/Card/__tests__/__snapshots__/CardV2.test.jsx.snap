// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing Card v2 Styles Uncontrolled renders CardContent correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "CardV2__CardContent-mom78k-2",
        "isStatic": false,
        "rules": Array [
          [Function],
          " color:",
          [Function],
          ";flex-grow:1;padding:24px;",
        ],
      },
      "displayName": "CardV2__CardContent",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "CardV2__CardContent-mom78k-2",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Card v2 Styles Uncontrolled renders CardFooter correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "CardV2__CardFooter-mom78k-3",
        "isStatic": false,
        "rules": Array [
          [Function],
          " font-family:",
          [Function],
          ";color:",
          [Function],
          ";border-top:1px solid ",
          [Function],
          ";width:100%;height:48px;display:flex;align-items:center;justify-content:center;text-align:center;",
        ],
      },
      "displayName": "CardV2__CardFooter",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "CardV2__CardFooter-mom78k-3",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Card v2 Styles Uncontrolled renders InnerCard correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "CardV2__InnerCard-mom78k-4",
        "isStatic": false,
        "rules": Array [
          [Function],
          " border-radius:4px;background-color:",
          [Function],
          ";min-width:50px;min-height:50px;display:flex;flex-direction:column;justify-content:flex-start;align-items:center;",
          "box-shadow:unset;border:1px solid ",
          [Function],
          ";background-color:",
          [Function],
          ";",
        ],
      },
      "displayName": "CardV2__InnerCard",
      "foldedComponentIds": Array [
        "CardV2__Card-mom78k-0",
      ],
      "render": [Function],
      "styledComponentId": "CardV2__InnerCard-mom78k-4",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Card v2 Styles Uncontrolled renders InnerCardHeader correctly 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "CardV2__InnerCardHeader-mom78k-5",
        "isStatic": false,
        "rules": Array [
          [Function],
          " font-family:",
          [Function],
          ";color:",
          [Function],
          ";border-bottom:1px solid ",
          [Function],
          ";width:100%;height:48px;display:flex;align-items:center;justify-content:center;text-align:center;padding:0px 24px;box-sizing:border-box;position:relative;&:after{content:\\"\\";position:absolute;bottom:-1px;height:0px;left:24px;width:calc(100% - 48px);border-bottom:2px solid ",
          [Function],
          ";}",
          "justify-content:left;text-align:flex-start;font-weight:",
          [Function],
          ";",
        ],
      },
      "displayName": "CardV2__InnerCardHeader",
      "foldedComponentIds": Array [
        "CardV2__CardHeader-mom78k-1",
      ],
      "render": [Function],
      "styledComponentId": "CardV2__InnerCardHeader-mom78k-5",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Card v2 Styles controlled Card renders correctly without props 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "CardV2__Card-mom78k-0",
        "isStatic": false,
        "rules": Array [
          [Function],
          " border-radius:4px;background-color:",
          [Function],
          ";min-width:50px;min-height:50px;display:flex;flex-direction:column;justify-content:flex-start;align-items:center;",
        ],
      },
      "displayName": "CardV2__Card",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "CardV2__Card-mom78k-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Card v2 Styles controlled Card renders correctly without with elevation=1 1`] = `
<StyledComponent
  elevation="1"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "CardV2__Card-mom78k-0",
        "isStatic": false,
        "rules": Array [
          [Function],
          " border-radius:4px;background-color:",
          [Function],
          ";min-width:50px;min-height:50px;display:flex;flex-direction:column;justify-content:flex-start;align-items:center;",
        ],
      },
      "displayName": "CardV2__Card",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "CardV2__Card-mom78k-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Card v2 Styles controlled Card renders correctly without with elevation=2 1`] = `
<StyledComponent
  elevation="2"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "CardV2__Card-mom78k-0",
        "isStatic": false,
        "rules": Array [
          [Function],
          " border-radius:4px;background-color:",
          [Function],
          ";min-width:50px;min-height:50px;display:flex;flex-direction:column;justify-content:flex-start;align-items:center;",
        ],
      },
      "displayName": "CardV2__Card",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "CardV2__Card-mom78k-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Card v2 Styles controlled Card renders correctly without with elevation=3 1`] = `
<StyledComponent
  elevation="3"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "CardV2__Card-mom78k-0",
        "isStatic": false,
        "rules": Array [
          [Function],
          " border-radius:4px;background-color:",
          [Function],
          ";min-width:50px;min-height:50px;display:flex;flex-direction:column;justify-content:flex-start;align-items:center;",
        ],
      },
      "displayName": "CardV2__Card",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "CardV2__Card-mom78k-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Card v2 Styles controlled Card renders correctly without with elevation=other 1`] = `
<StyledComponent
  elevation="other"
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "CardV2__Card-mom78k-0",
        "isStatic": false,
        "rules": Array [
          [Function],
          " border-radius:4px;background-color:",
          [Function],
          ";min-width:50px;min-height:50px;display:flex;flex-direction:column;justify-content:flex-start;align-items:center;",
        ],
      },
      "displayName": "CardV2__Card",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "CardV2__Card-mom78k-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Card v2 Styles controlled CardHeader renders correctly without props 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "CardV2__CardHeader-mom78k-1",
        "isStatic": false,
        "rules": Array [
          [Function],
          " font-family:",
          [Function],
          ";color:",
          [Function],
          ";border-bottom:1px solid ",
          [Function],
          ";width:100%;height:48px;display:flex;align-items:center;justify-content:center;text-align:center;padding:0px 24px;box-sizing:border-box;position:relative;&:after{content:\\"\\";position:absolute;bottom:-1px;height:0px;left:24px;width:calc(100% - 48px);border-bottom:2px solid ",
          [Function],
          ";}",
        ],
      },
      "displayName": "CardV2__CardHeader",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "CardV2__CardHeader-mom78k-1",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`Testing Card v2 Styles controlled CardHeader renders correctly without with highlight=alert 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "CardV2__Card-mom78k-0",
        "isStatic": false,
        "rules": Array [
          [Function],
          " border-radius:4px;background-color:",
          [Function],
          ";min-width:50px;min-height:50px;display:flex;flex-direction:column;justify-content:flex-start;align-items:center;",
        ],
      },
      "displayName": "CardV2__Card",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "CardV2__Card-mom78k-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  highlight="alert"
/>
`;

exports[`Testing Card v2 Styles controlled CardHeader renders correctly without with highlight=dark 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "CardV2__Card-mom78k-0",
        "isStatic": false,
        "rules": Array [
          [Function],
          " border-radius:4px;background-color:",
          [Function],
          ";min-width:50px;min-height:50px;display:flex;flex-direction:column;justify-content:flex-start;align-items:center;",
        ],
      },
      "displayName": "CardV2__Card",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "CardV2__Card-mom78k-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  highlight="dark"
/>
`;

exports[`Testing Card v2 Styles controlled CardHeader renders correctly without with highlight=default 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "CardV2__Card-mom78k-0",
        "isStatic": false,
        "rules": Array [
          [Function],
          " border-radius:4px;background-color:",
          [Function],
          ";min-width:50px;min-height:50px;display:flex;flex-direction:column;justify-content:flex-start;align-items:center;",
        ],
      },
      "displayName": "CardV2__Card",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "CardV2__Card-mom78k-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  highlight="default"
/>
`;

exports[`Testing Card v2 Styles controlled CardHeader renders correctly without with highlight=error 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "CardV2__Card-mom78k-0",
        "isStatic": false,
        "rules": Array [
          [Function],
          " border-radius:4px;background-color:",
          [Function],
          ";min-width:50px;min-height:50px;display:flex;flex-direction:column;justify-content:flex-start;align-items:center;",
        ],
      },
      "displayName": "CardV2__Card",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "CardV2__Card-mom78k-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  highlight="error"
/>
`;

exports[`Testing Card v2 Styles controlled CardHeader renders correctly without with highlight=other 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "CardV2__Card-mom78k-0",
        "isStatic": false,
        "rules": Array [
          [Function],
          " border-radius:4px;background-color:",
          [Function],
          ";min-width:50px;min-height:50px;display:flex;flex-direction:column;justify-content:flex-start;align-items:center;",
        ],
      },
      "displayName": "CardV2__Card",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "CardV2__Card-mom78k-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  highlight="other"
/>
`;

exports[`Testing Card v2 Styles controlled CardHeader renders correctly without with highlight=primary 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "CardV2__Card-mom78k-0",
        "isStatic": false,
        "rules": Array [
          [Function],
          " border-radius:4px;background-color:",
          [Function],
          ";min-width:50px;min-height:50px;display:flex;flex-direction:column;justify-content:flex-start;align-items:center;",
        ],
      },
      "displayName": "CardV2__Card",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "CardV2__Card-mom78k-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  highlight="primary"
/>
`;

exports[`Testing Card v2 Styles controlled CardHeader renders correctly without with highlight=success 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "CardV2__Card-mom78k-0",
        "isStatic": false,
        "rules": Array [
          [Function],
          " border-radius:4px;background-color:",
          [Function],
          ";min-width:50px;min-height:50px;display:flex;flex-direction:column;justify-content:flex-start;align-items:center;",
        ],
      },
      "displayName": "CardV2__Card",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "CardV2__Card-mom78k-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
  highlight="success"
/>
`;
